pipeline/git_preflight.go

```
@@ -0,0 +1,709 @@
+//   - safetyLatches(workDir) passes
+//   - workdir is empty (no files outside any pre-existing `.git`) —
+//     see workdirHasContent for the rationale (we refuse rather than
+//     auto-`git add -A` user files that might be secrets / artifacts)
+//
+// Returns a wrapped ErrGitAutoInitRefused if any latch fires.
+func runAutoInit(ctx context.Context, workDir string, allowInit bool, interactive bool, promptYN func(prompt string) bool) error {
+	// Latch 1: explicit consent. --allow-init is required in non-interactive
+	// mode. In interactive mode, the [Y/n] prompt substitutes.
+	if !allowInit {
+		if !interactive {
+			return fmt.Errorf("%w: --git=init requires --allow-init in non-interactive runs.\n\n  Pass --allow-init to acknowledge that tracker may run `git init` in this directory, or run `git init` manually first.", ErrGitAutoInitRefused)
```

golangci-lint (2.12.2)

[error] ST1005: error strings should not end with punctuation or newlines

(staticcheck)